src/storage/src/table/batch_table/storage_table.rs

```diff
@@ -241,31 +238,11 @@ impl<S: StateStore> StorageTableInner<S, EitherSerde> {
         let row_serde = {
             let schema = Arc::from(data_types.into_boxed_slice());
             if versioned {
-                let column_with_default = table_columns
-                    .iter()
-                    .enumerate()
-                    .filter(|(_, c)| c.is_default())
-                    .map(|(i, c)| {
-                        if let GeneratedOrDefaultColumn::DefaultColumn(DefaultColumnDesc { expr }) =
-                            c.generated_or_default_column.clone().unwrap()
-                        {
-                            (
-                                i,
-                                build_from_prost(&expr.expect("expr should not be none"))
-                                    .expect("build_from_prost error")
-                                    .eval_row_infallible(&OwnedRow::empty(), |_err| {})
-                                    .now_or_never()
-                                    .expect("constant expression should not be async"),
-                            )
-                        } else {
-                            unreachable!()
-                        }
-                    });
-                let mut serde = ColumnAwareSerde::new(&column_ids, schema);
-                serde.set_default_columns(column_with_default);
+                let serde =
+                    ColumnAwareSerde::new(&column_ids, schema, table_columns.iter().cloned());
```

Comment:
Oops. Suddenly I realize that we should also take `value_indices` into consideration (just like how we get `column_ids` and `schema`). Though it seems not possible to both have `value_indices` set to partial columns and serde method to `ColumnAware`... 

In that case, maybe we can directly pass the `table_columns` and `value_indices` in to resolve everything?